File: backend/systempus/src/main/java/br/com/systempus/systempus/domain/Modulo.java
package br.com.systempus.systempus.domain;

import java.time.LocalDate;
import java.util.List;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "modulo")
public class Modulo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull(message = "Campo Obrigatório")
    private String nome;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataInicio;
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataFim;

    @ManyToOne
    @JsonBackReference(value = "curso_modulos")
    @JoinColumn(name = "curso_id")
    private Curso curso;

    // @JsonManagedReference(value = "disciplina_modulo")
    @OneToMany(mappedBy = "modulo")
    private List<Disciplina> disciplinas;


}

File: app_flutter_horario_mobile/lib/domain/modulo.dart
import 'package:app_flutter_horario_mobile/domain/curso.dart';
import 'package:app_flutter_horario_mobile/domain/disciplina.dart';

class Modulo {
  int? _id;
  int? _numero;
  DateTime? _dataInicio;
  DateTime? _dataFim;
  Curso? _curso;
  List<Disciplina>? _disciplinas;

  Modulo(
      {int? id,
      int? numero,
      DateTime? dataInicio,
      DateTime? dataFim,
      Curso? curso,
      List<Disciplina>? disciplinas});

  int? get id => this._id;

  set id(int? value) => this._id = value;

  get numero => this._numero;

  set numero(value) => this._numero = value;

  get dataInicio => this._dataInicio;

  set dataInicio(value) => this._dataInicio = value;

  get dataFim => this._dataFim;

  set dataFim(value) => this._dataFim = value;

  get curso => this._curso;

  set curso(value) => this._curso = value;

  get disciplinas => this._disciplinas;

  set disciplinas(value) => this._disciplinas = value;
}
